### PR TITLE
Hide package metadata when the package is deleted

### PR DESCRIPTION
**Pull request by joelverhagen**
Fix https://github.com/NuGet/NuGetGallery/issues/4717.

I recommend ignoring whitespace in the diff:
https://github.com/NuGet/NuGetGallery/pull/4978/files?w=1

This also add a default warning banner for local development (like DEV and INT).

## Before

screencapture-localhost-packages-nuget-server-2-14-0-1510258411423

## After

screencapture-localhost-packages-nuget-server-2-14-0-1510258828190

### CODE REVIEW

**Comment by joelverhagen** [comment]
I disagree. These metadata values are essentially dead. It would add complexity to the code and provides no administrative/owner benefit.

**Comment by scottbommarito** [comment]
They're dead, but I think they could help admins and owners understand what was deleted in case a package was deleted accidentally or a user has forgotten what package was deleted.

I suppose it's not too big of a deal, however.

**Comment by anangaur** [comment]
The metadata info is irrelevant once the package is deleted. The only information to persist would be to show the owner of the package that the package was deleted.

File: src/NuGetGallery/Web.config [hunk 1]
```diff
@@ -56,7 +56,7 @@
     <add key="Gallery.Environment" value="Development" />
     
     <!-- banner warning on the top - leave blank to remove the banner from the site. -->
-    <add key="Gallery.WarningBanner" value="" />
+    <add key="Gallery.WarningBanner" value="This is the local development environment." />
```
Comment (loic-sharma) [inline src/NuGetGallery/Web.config]: Is this change intended to be checked-in? Personally I would prefer to not see this while developing.

Comment (joelverhagen) [reply src/NuGetGallery/Web.config]: I think it's a good practice to have it. When using the delete UI or upload, it should be very clear which environment you are working on.

If you feel strongly I can remove it.

Comment (skofman1) [reply src/NuGetGallery/Web.config]: Just make sure our prod config overrides this value explicitly 😆
